src/python/test/test_extract_dataset.py

```diff
@@ -0,0 +1,134 @@
+        cmd = ["dx", "extract_dataset", cohort_record, "--fields", "patient.patient_id" , ",", "patient.name", ",", "patient.weight", ",",
+               "patient.date_of_birth", ",", "patient.verified_dtm", ",", "test.test_id", ",", "trial_visit.visit_id", ",", "baseline.baseline_id", 
+               ",", "hospital.hospital_id", ",", "doctor.doctor_id","--sql", "-o", "-"]
+        process = subprocess.Popen(cmd, stdout=subprocess.PIPE, universal_newlines=True)
+        stdout = process.communicate()[0]
+        self.assertTrue(truth_output==stdout.strip())
+
+    def test_e2e_dataset_fields(self):
+        dataset_record = "project-G9j1pX00vGPzF2XQ7843k2Jq:record-GBGpKZ80G8j8k4G84P7GP0B5"
+        out_directory = tempfile.mkdtemp()
+        cmd = ["dx", "extract_dataset", dataset_record, "--fields", "patient.patient_id" , ",", "patient.name", ",", "patient.weight", ",",
+               "patient.date_of_birth", ",", "patient.verified_dtm", ",", "test.test_id", ",", "trial_visit.visit_id", ",", "baseline.baseline_id", 
+               ",", "hospital.hospital_id", ",", "doctor.doctor_id", "-o", out_directory]
+        subprocess.check_call(cmd)
+        truth_file = "project-G9j1pX00vGPzF2XQ7843k2Jq:file-GBGpx800vGPy4qf84JJQBX3x"
+        self.end_to_end_fields(out_directory=out_directory, rec_name = "extract_dataset_test.csv", truth_file=truth_file)
+
+    def test_e2e_cohortbrowser_fields(self):
+        cohort_record = "project-G9j1pX00vGPzF2XQ7843k2Jq:record-GBGq1pQ0vGPf0qYg30Jg7kkG"
+        out_directory = tempfile.mkdtemp()
+        cmd = ["dx", "extract_dataset", cohort_record, "--fields", "patient.patient_id" , ",", "patient.name", ",", "patient.weight", ",",
+               "patient.date_of_birth", ",", "patient.verified_dtm", ",", "test.test_id", ",", "trial_visit.visit_id", ",", "baseline.baseline_id", 
+               ",", "hospital.hospital_id", ",", "doctor.doctor_id", "-o", out_directory]
+        subprocess.check_call(cmd)
+        truth_file = "project-G9j1pX00vGPzF2XQ7843k2Jq:file-GBGq7FQ0vGPpK3GbF0Xbjz17"
+        self.end_to_end_fields(out_directory=out_directory, rec_name = "Combined_Cohort_Test.csv", truth_file=truth_file)
+    
+    def test_file_already_exists(self):
+        cohort_record = "project-G9j1pX00vGPzF2XQ7843k2Jq:record-GBGq1pQ0vGPf0qYg30Jg7kkG"
+        out_directory = tempfile.mkdtemp()
+        open(os.path.join(out_directory, "Combined_Cohort_Test.csv"), 'w').close()
+        cmd = ["dx", "extract_dataset", cohort_record, "--fields", "patient.patient_id" , ",", "patient.name", "-o", out_directory]
+        output = subprocess.run(cmd, capture_output=True, text=True)
+        self.assertTrue("Error: path already exists" in output.stderr)
+        shutil.rmtree(out_directory)
+
+    def end_to_end_ddd(self, out_directory, rec_name):
```

Comment:
what does `ddd` stand for? Please rename the method to be more readable, thanks!

Reply:
Dump Dataset Dictionary.
`-ddd/--dump-dataset-dictionary` is one of the arguments that can be passed with `dx extract_dataset` command.